Conch/lib/Conch/Controller/Orc/Lifecycles.pm

```diff
@@ -0,0 +1,128 @@
+=head1 NAME
+
+Conch::Controller::Orc::Lifecycles
+
+=head1 METHODS
+
+=cut
+
+package Conch::Controller::Orc::Lifecycles;
+
+use Mojo::Base 'Mojolicious::Controller', -signatures;
+
+use Conch::Orc;
+
+
+=head2 get_all
+
+Get all lifecycles
+
+=cut
+
+sub get_all ($c) {
+	return $c->status_with_validation(200, 
+		OrcLifecycles => Conch::Orc::Lifecycle->all()
+	);
+}
+
+
+=head2 get_one
+
+Get a single lifecycle
+
+=cut
+
+sub get_one ($c) {
+	my $l = Conch::Orc::Lifecycle->from_id($c->param('id'));
+	return $c->status(404 => { error => "Not found" }) unless $l;
+	return $c->status(404 => { error => "Not found" }) if $l->deactivated;
+
+	$c->status_with_validation(200, OrcLifecycle => $l );
+}
+
+
+=head2 create
+
+Create a new lifecycle
+
+=cut
+
+sub create ($c) {
+	return $c->status(403) unless $c->is_global_admin;
+	my $body = $c->validate_input('OrcLifecycleCreate') or return;
+	
+	my $l = Conch::Orc::Lifecycle->new($body->%*)->save();
```

Comment:
You should check that the supplied name & version don't exist before attempting to create to avoid throwing 500s.